### ISSUE TITLE
[NEXT-665] TypeScript error when using generateMetadata() in WebStorm IDE

### ISSUE THREAD

**Issue by patrykszwed** [vercel/next.js#46431]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.13.0
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.2.2-canary.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Metadata (metadata, generateMetadata, next/head, head.js), TypeScript

### Link to the code that reproduces this issue

https://github.com/patrykszwed/next-metadata-ts-error

### To Reproduce

1. Run `npx create-next-app --example reproduction-template-app-dir reproduction-app`
2. Add `generateMetadata` function to `page.tsx` at a root level. That code should be enough:

```js
export async function generateMetadata() {
  return {};
}

export default function Home() {
  return null
}
```

### Describe the Bug

When implementing a `generateMetadata` function and opening a project in WebStorm, a TypeScript error is shown at a line where the function is defined. The error says: `TS1055: Type 'Metadata | Promise ' is not a valid async function return type in ES5/ES3 because it does not refer to a Promise-compatible constructor value.`

However, building the project works fine. Moreover, when opening the same project in Visual Studio Code, there are no errors at all.  

WebStorm:
![image](https://user-images.githubusercontent.com/32751479/221384345-0a281c9f-4c85-42be-a024-b1405e8f93b3.png)

Visual Studio Code:
![image](https://user-images.githubusercontent.com/32751479/221384351-de0cc88f-183d-4748-976e-f0cfa5648e0d.png)

### Expected Behavior

No TypeScript errors are shown when defining a valid `generateMetadata` function.

<sub>[NEXT-665](https://linear.app/vercel/issue/NEXT-665/typescript-error-when-using-generatemetadata-in-webstorm-ide)</sub>

**Comment by t3dotgg** [comment]
Confirming this isn't Webstorm specific, appears these typedefs are bad
![image](https://user-images.githubusercontent.com/6751787/222047365-e6bddc5e-4633-415e-9372-4f842e4d5b7b.png)

https://github.com/vercel/next.js/pull/45723/files#r1121140364

**Comment by ijjk** [comment]
This should be fixed by https://github.com/vercel/next.js/pull/46596 which is now available in `v13.3.3-canary.0`
